lib/data.js

```diff
@@ -0,0 +1,23 @@
+import base from 'taskcluster-base';
+
+var data = {} = module.exports;
+
+data.SecretEntity = function(cfg) {
+  return base.Entity.configure({
+    version:          1,
+    signEntities:     true,
+    partitionKey:     base.Entity.keys.ConstantKey('secrets'),
+    rowKey:           base.Entity.keys.StringKey('name'),
+    properties: {
+      name:           base.Entity.types.String,
+      secret:         base.Entity.types.EncryptedJSON,
+      expires:        base.Entity.types.Date
+    }
+  }).setup({
+    account:          cfg.get('azure:accountName'),
+    credentials:      cfg.get('taskcluster:credentials'),
+    table:            cfg.get('azure:tableName'),
+    cryptoKey:        cfg.get('azure:cryptoKey'),
+    signingKey:       cfg.get('azure:signingKey')
+  });
+};
```

Comment:
This is a nit, I'm just explaining the reasoning behind the way `base.Entity`.
```js
// Step 1 declare the entity layout (table schema)
// This is validated to be somewhat sane when loaded, ie. before you even start the main function
let Secret = base.Entity.configure({
  version:          1,
  signEntities:     true,
  partitionKey:     base.Entity.keys.ConstantKey('secrets'),
  rowKey:           base.Entity.keys.StringKey('name'),
  properties: {
    name:           base.Entity.types.String,
    secret:         base.Entity.types.EncryptedJSON,
    expires:        base.Entity.types.Date
  }
});

// Step 2: setup Secret with configuration, this stuff happens inside the main function and
// may fail at runtime...
data.SecretEntity = function(cfg) {
  return Secret.setup({
    account:          cfg.get('azure:accountName'),
    credentials:      cfg.get('taskcluster:credentials'),
    table:            cfg.get('azure:tableName'),
    cryptoKey:        cfg.get('azure:cryptoKey'),
    signingKey:       cfg.get('azure:signingKey')
  });
};
```

If were coding in say C++ we would say that bugs in step (1) would cause compile-time errors, where as bugs in step (2) would cause runtime errors.
This design goes for `base.Exchanges` and `base.API` too, I'm basically designing such that you get simple errors as soon as possible (at declaration time), rather than you getting some obscure error with a giant stack-trace at runtime.

Follow these guidelines and you'll get a lot more errors up front for free..